src/components/ClientCard/ClientCard.jsx

```diff
@@ -4,38 +4,107 @@
   * TODO: style based on currently selected card.
 */
 
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import moment from 'moment';
 import { connect } from 'react-redux';
+
 import { setActiveChat, loadMessages } from '../../store/Chat';
+
 import './ClientCard.css';
 
-const ClientCard = props => (
-  <li className="ClientCard" >
-    <button
-      className="ClientCard__btn"
-      onClick={() => {
-        props.setActiveChat(props.chat);
-
-        props.loadMessages(props.config, props.chat.id);
-      }}
-    >
-      {props.children}
-    </button>
-  </li>
-);
-
-ClientCard.propTypes = {
-  chat: PropTypes.object.isRequired,
-  config: PropTypes.object.isRequired,
-  children: PropTypes.node.isRequired,
-  setActiveChat: PropTypes.func.isRequired,
-  loadMessages: PropTypes.func.isRequired,
-};
+// TODO: Improve this by pulling ping/pong from daemon's connection with the
+//   client.
+// If a client is not active for 2 minutes, consider them offline
+const ONLINE_TIMEOUT = 120000;
+
+class ClientCard extends Component {
+  static propTypes = {
+    chat: PropTypes.object.isRequired,
+    messages: PropTypes.array.isRequired,
+    config: PropTypes.object.isRequired,
+    children: PropTypes.node.isRequired,
+    setActiveChat: PropTypes.func.isRequired,
+    loadMessages: PropTypes.func.isRequired,
+    activeId: PropTypes.string,
+  }
+
+  getChatLastMessage () {
+    const { messages, chat, activeId } = this.props;
+    let filteredMessages = messages.filter(msg => (
+      msg.chat === chat.id && msg.author === `client.${msg.chat}`
+    ));
+
+    if (filteredMessages.length > 0) {
+      return filteredMessages[filteredMessages.length - 1];
+    }
+
+    return false;
+  }
+
+  select () {
+
+  }
+
+  render () {
+    const { chat, config, activeId } = this.props;
+    const lastMessage = this.getChatLastMessage();
+    const classes = [
+      'ClientCard',
+      (activeId === chat.id) ? 'active' : '',
+    ];
+    const statusClasses = [
+      'ClientCard__status',
+      (lastMessage.timestamp &&
+        (new Date(lastMessage.timestamp) >= new Date().getTime() - ONLINE_TIMEOUT)) ?
+      'online' :
+      'offline',
```

Comment:
this line is really ugly. I'd recommend making it into a method that returns online/offline status.

Reply:
oof @broneks TELLING IT LIKE ITISSSSSS

Reply:
but yeah we should do that

Reply:
I definitely was tired and wanted this to go in asap, but you're right @broneks, will clean up